pkg/crypto/crypto_test.go

```diff
@@ -111,3 +112,85 @@ func decryptConfigMap(configMap string, keyPath string, noncePath string) string
 
 	return string(plaintextBytes)
 }
+
+
+func TestDecryptConfigMapWrongkey(t *testing.T)  {
+
+        /*Test Objective : Negative test: In Symmetric encryption only one key (a secret key) is used to
+         both encrypt and decrypt. If different key is passed for decrypt, decrypt function should fail.
+         In this test, wrong key value is being used for decryption, test should FAIL*/
+
+        testString := "Sample String to Test Symmetric Encryption"
+
+        dir1, err := ioutil.TempDir("", "TestEncryption-2")
+        if err != nil {
+                t.Error(err)
+        }
+        defer os.RemoveAll(dir1)
+
+        keyPath := dir1 + "symmKeyFile"
+        wrongkeyPath := dir1 + "wrongsymmKeyFile"
+
+        key := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, key); err != nil {
+                t.Error(err)
+        }
+
+        wrongkey := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, wrongkey); err != nil {
+                t.Error(err)
+        }
+
+
+        //Encode the Key
+        keyEnc := b64.StdEncoding.EncodeToString(key)
+        if err != nil {
+                t.Error(err)
+        }
+
+        //encode wrong key
+        wrongkeyEnc := b64.StdEncoding.EncodeToString(wrongkey)
+        if err != nil {
```

Comment:
Which error is this? `EncodeToString` doesn't return error. https://golang.org/pkg/encoding/base64/#example_Encoding_EncodeToString

Reply:
noted. will correct this. this is followed in existing bucket. will correct them too.
//Encode the Key
	keyEnc := b64.StdEncoding.EncodeToString(key)
	if err != nil {
		t.Error(err)
	}

Reply:
Also, Do you prefer I address this in this PR or create a separate PR for this? Please suggest.

Reply:
@harche: Please help with above query.

Reply:
I am not sure if this change is required at all.

Reply:
let's first understand if the modifying the `crypto.go` so that an unit test passes is a good idea. I would like to see the results of the actual deployment.